src/drivers/interrupt/cortexm_irq_handle.S

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file
+ * @brief
+ *
+ * @author  Alex Kalmuk
+ * @date    27.03.2017
+ */
+
+.text
+.thumb
+.syntax unified
+.global interrupt_handle_enter
+interrupt_handle_enter:
+
+    mov    r0, sp
+    sub    sp, #58
+    str    r0, [sp, #56]
+    stmia  sp, {r0 - r12}
+    str    lr, [sp, #52]
+
+    mov    r0, sp
+    bl     interrupt_handle
+
+.text
+.thumb
+.syntax unified
+.global __irq_trampoline
+__irq_trampoline:
+
+    cpsid  i
+    # r0 is a pointer to struct irq_saved_state
+    ldr    sp,  [r0, #32]
+    ldr    r14, [r0, #36]
+    bx     r14
```

Comment:
Got it, was confused by ctx->lr asignment just above. Ok, looks correct.
If you don't mind, I'd recommend to change __irq_trampoline interface: 2nd arg not used, only sp and restore lr used from 1st. If you just pass return sp and lr, then __irq_trampoline could be just 
```
mov sp, r0
bx r1
```